code/model/SiteTreeLinkTracking.php

```diff
@@ -28,12 +28,39 @@
  */
 class SiteTreeLinkTracking extends DataExtension {
 
-	public $parser;
+	/**
+	 * @var SiteTreeLinkTracking_Parser
+	 */
+	protected $parser;
 
+	/**
+	 * Inject parser for each page
+	 *
+	 * @var array
+	 * @config
+	 */
 	private static $dependencies = array(
-		'parser' => '%$SiteTreeLinkTracking_Parser'
+		'Parser' => '%$SiteTreeLinkTracking_Parser'
 	);
 
+	/**
+	 * Parser for link tracking
+	 *
+	 * @return SiteTreeLinkTracking_Parser
+	 */
+	public function getParser() {
```

Comment:
This was a work-around for virtual pages trying to virtualise the parser; Having an explicit getter / setter prevented this.